websockets/blob.go

```diff
@@ -0,0 +1,178 @@
+						err = fmt.Errorf("could not fetch Blob data: %w", err)
+						break
+					}
+					_, err = b.data.Write(bp.data.Bytes())
+				default:
+					err = fmt.Errorf("unsupported type: %T", part)
+				}
+			default:
+				err = fmt.Errorf("unsupported type: %T", part)
+			}
+			if err != nil {
+				common.Throw(rt, fmt.Errorf("failed to process [blobParts]: %w", err))
+			}
+		}
+	}
+
+	if len(call.Arguments) > 1 && !sobek.IsUndefined(call.Arguments[1]) {
+		opts := call.Arguments[1]
+		if !isObject(opts) {
+			common.Throw(rt, errors.New("[options] must be an object"))
+		}
+
+		typeOpt := opts.ToObject(rt).Get("type")
+		if !sobek.IsUndefined(typeOpt) {
+			b.typ = typeOpt.String()
+		}
+	}
+
+	obj := rt.NewObject()
+	must(rt, obj.DefineAccessorProperty("_ref", rt.ToValue(func() sobek.Value {
+		return rt.ToValue(b)
+	}), nil, sobek.FLAG_FALSE, sobek.FLAG_FALSE)) // internal use only
+	must(rt, obj.DefineAccessorProperty("size", rt.ToValue(func() sobek.Value {
+		return rt.ToValue(b.data.Len())
+	}), nil, sobek.FLAG_FALSE, sobek.FLAG_TRUE))
+	must(rt, obj.DefineAccessorProperty("type", rt.ToValue(func() sobek.Value {
+		return rt.ToValue(b.typ)
+	}), nil, sobek.FLAG_FALSE, sobek.FLAG_TRUE))
+
+	must(rt, obj.Set("arrayBuffer", func(_ sobek.FunctionCall) sobek.Value {
+		promise, resolve, _ := rt.NewPromise()
+		resolve(rt.NewArrayBuffer(b.data.Bytes()))
+		return rt.ToValue(promise)
+	}))
+	must(rt, obj.Set("bytes", func(_ sobek.FunctionCall) sobek.Value {
+		promise, resolve, reject := rt.NewPromise()
+		data, err := rt.New(rt.Get("Uint8Array"), rt.ToValue(b.data.Bytes()))
+		if err == nil {
+			resolve(data)
+		} else {
+			reject(fmt.Errorf("failed to create Uint8Array: %w", err))
+		}
+		return rt.ToValue(promise)
+	}))
+	must(rt, obj.Set("slice", func(call sobek.FunctionCall) sobek.Value {
+		from, to, ct := 0, b.data.Len(), ""
+		if len(call.Arguments) > 0 {
+			from = int(call.Arguments[0].ToInteger())
+		}
+		if len(call.Arguments) > 1 {
+			to = int(call.Arguments[1].ToInteger())
+			if to < 0 {
+				to = b.data.Len() + to
+			}
+		}
+		if len(call.Arguments) > 2 {
+			ct = call.Arguments[2].String()
+		}
+
+		opts := rt.NewObject()
+		must(rt, opts.Set("type", ct))
+
+		slice, err := rt.New(rt.Get("Blob"), rt.ToValue([]interface{}{b.data.Bytes()[from:to]}), opts)
+		must(rt, err)
+		return slice
+	}))
+	must(rt, obj.Set("text", func(_ sobek.FunctionCall) sobek.Value {
+		promise, resolve, _ := rt.NewPromise()
+		resolve(b.text())
+		return rt.ToValue(promise)
+	}))
+	must(rt, obj.Set("stream", func(_ sobek.FunctionCall) sobek.Value {
+		return rt.ToValue(streams.NewReadableStreamFromReader(r.vu, &b.data))
+	}))
+
+	proto := call.This.Prototype()
+	must(rt, proto.Set("toString", func(_ sobek.FunctionCall) sobek.Value {
+		return rt.ToValue("[object Blob]")
+	}))
+	must(rt, obj.SetPrototype(proto))
+
+	return obj
+}
+
+// toByteSlice converts a slice of numbers to a slice of bytes.
+func toByteSlice(data interface{}) []byte {
```

Comment:
I'm open for suggestions, although none of the ones I've seen so far really convinces me. If you prefer any other, just tell me.